subprojects/language-java/src/main/java/org/gradle/external/javadoc/CoreJavadocOptions.java

```diff
@@ -61,45 +90,60 @@ public CoreJavadocOptions() {
     protected CoreJavadocOptions(JavadocOptionFile optionFile) {
         this.optionFile = optionFile;
 
-        overview = addStringOption("overview");
-        memberLevel = addEnumOption("memberLevel");
-        doclet = addStringOption("doclet");
-        docletpath = addPathOption("docletpath");
-        source = addStringOption("source");
-        classpath = addPathOption("classpath");
-        modulePath = addPathOption("-module-path");
-        sourcePath = addPathOption("-source-path");
-        bootClasspath = addPathOption("bootclasspath");
-        extDirs = addPathOption("extdirs");
-        outputLevel = addEnumOption("outputLevel", JavadocOutputLevel.QUIET);
-        breakIterator = addBooleanOption("breakiterator");
-        locale = addStringOption("locale");
-        encoding = addStringOption("encoding");
+        overview = addStringOption(OPTION_OVERVIEW);
+        memberLevel = addEnumOption(OPTION_MEMBERLEVEL);
+        doclet = addStringOption(OPTION_DOCLET);
+        docletpath = addPathOption(OPTION_DOCLETPATH);
+        source = addStringOption(OPTION_SOURCE);
+        classpath = addPathOption(OPTION_CLASSPATH);
+        modulePath = addPathOption(OPTION_MODULE_PATH);
+        sourcePath = addPathOption(OPTION_SOURCE_PATH);
+        bootClasspath = addPathOption(OPTION_BOOTCLASSPATH);
+        extDirs = addPathOption(OPTION_EXTDIRS);
+        outputLevel = addEnumOption(OPTION_OUTPUTLEVEL, JavadocOutputLevel.QUIET);
+        breakIterator = addBooleanOption(OPTION_BREAKITERATOR);
+        locale = addStringOption(OPTION_LOCALE);
+        encoding = addStringOption(OPTION_ENCODING);
 
         sourceNames = optionFile.getSourceNames();
+
+        knownCoreOptionNames = Collections.unmodifiableSet(new HashSet<>(optionFile.getOptions().keySet()));
```

Comment:
I would just use `ImmutableSet.copyOf` here -- no need for an extra wrapper.
```suggestion
        knownCoreOptionNames = ImmutableSet.copyOf(optionFile.getOptions().keySet());
```

Reply:
These are external classes though, so I didn't want to serve Guava collections to consumers (extension classes for non-standard doclets?) accessing the field, per our previous discussions about this.